stdlib/influxdata/influxdb/from.go

```diff
@@ -123,446 +76,35 @@ func (s *FromProcedureSpec) Kind() plan.ProcedureKind {
 
 func (s *FromProcedureSpec) Copy() plan.ProcedureSpec {
 	ns := new(FromProcedureSpec)
-
-	ns.Bucket = s.Bucket
-	ns.BucketID = s.BucketID
-
-	ns.BoundsSet = s.BoundsSet
-	ns.Bounds = s.Bounds
-
-	ns.FilterSet = s.FilterSet
-	ns.Filter = s.Filter.Copy().(*semantic.FunctionExpression)
-
-	ns.DescendingSet = s.DescendingSet
-	ns.Descending = s.Descending
-
-	ns.LimitSet = s.LimitSet
-	ns.PointsLimit = s.PointsLimit
-	ns.SeriesLimit = s.SeriesLimit
-	ns.SeriesOffset = s.SeriesOffset
-
-	ns.WindowSet = s.WindowSet
-	ns.Window = s.Window
-
-	ns.GroupingSet = s.GroupingSet
-	ns.OrderByTime = s.OrderByTime
-	ns.GroupMode = s.GroupMode
-	ns.GroupKeys = s.GroupKeys
-
-	ns.AggregateSet = s.AggregateSet
-	ns.AggregateMethod = s.AggregateMethod
-
+	*ns = *s
 	return ns
 }
 
-// TimeBounds implements plan.BoundsAwareProcedureSpec
-func (s *FromProcedureSpec) TimeBounds(predecessorBounds *plan.Bounds) *plan.Bounds {
-	if s.BoundsSet {
-		bounds := &plan.Bounds{
-			Start: values.ConvertTime(s.Bounds.Start.Time(s.Bounds.Now)),
-			Stop:  values.ConvertTime(s.Bounds.Stop.Time(s.Bounds.Now)),
-		}
-		return bounds
-	}
-	return nil
-}
-
 func (s FromProcedureSpec) PostPhysicalValidate(id plan.NodeID) error {
-	if !s.BoundsSet || (s.Bounds.Start.IsZero() && s.Bounds.Stop.IsZero()) {
-		var bucket string
-		if len(s.Bucket) > 0 {
-			bucket = s.Bucket
-		} else {
-			bucket = s.BucketID
-		}
-		return fmt.Errorf(`%s: results from "%s" must be bounded`, id, bucket)
+	if !s.Bounded {
+		return fmt.Errorf(`%s: results from "%s" must be bounded`, id, s.Bucket)
```

Comment:
I wonder if this is needed here?  As it is now, if we have a `from` in the vanilla REPL, it will say it needs to be bounded.  Then when the user bounds it with a `range`, execution says, "I can't do that."  

Maybe vanilla `from` doesn't need to be bounded?  It will be an error to execute it no matter what.

Reply:
I kept a behavior similar to InfluxDB's from because some tests rely on it. I was tempted to remove the need for a bound. Should I do it and change tests as a consequence?

Reply:
👍 seems legit. I should change some tests around. Like the controller's ones that expect to fail if no range is applied.

Reply:
Oh, yeah.  I think I wrote that one.  Maybe you can use your valdiation-breaking rules to cause a planner error in that case.